firedrake/tsfc_interface.py

```diff
@@ -84,7 +84,7 @@ def _cache_store(cls, key, val):
             # No need for a barrier after this, since non root
             # processes will never race on this file.
             with gzip.open(tempfile, 'wb') as f:
-                cPickle.dump(val, f)
+                cPickle.dump(val, f, 0)
```

Comment:
Problem here is that some `__init__`/`__new__` overrides don't work well with higher protocol versions.